### PR TITLE
update react some component

### PR DESCRIPTION
**Pull request by canyugs**

### CODE REVIEW

File: web/src/components/HotTopic.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          <Button type ="submit">{actionText}</Button>
+        </form>
+      </Info>
+      <Image src={imageSrc} />
+    </Container>
+  )
+}
+
+HotTopic.defaultProps = {
+  title: "駭客狗狗",
```
Comment (gcobcqwe) [inline web/src/components/HotTopic.js:70]: XD

File: web/src/components/User.js [hunk 2]
```diff
@@ -1,130 +1,263 @@
...
-      <PointWrapper>
-        <Quantity>{points}</Quantity>
-        <Unit>P</Unit>
-      </PointWrapper>
-
+    </Container>
+    </PositionFixed>
+    <UserWide>
+      <Username>{nickname}</Username>
+      <UserSignout href="">sign out</UserSignout>
```
Comment (gcobcqwe) [inline web/src/components/User.js:245]: Remove sign out. We use token-based and it doesn't like username/password have the sign-out function.

File: web/src/index.js [hunk 3]
```diff
@@ -58,20 +60,25 @@ const App = () => {
     }).catch((error) => {
       setAuthorized(false);
     });
-  },[token]); // monitor token change then trigger this
+  },[]); // monitor token change then trigger this
```
Comment (gcobcqwe) [inline web/src/index.js:63]: Looks good for using react hook and state. 👍 
We need to fetch token from the query string first and if null then fetch the local cookie variable, right? Store the token into Cookie.
TODO:  Update setUser response data format.
I modified user API SEPC, will deploy it to the staging site (TBD), need to update setUser response data format.
https://docs.google.com/document/d/1qnv3KIdgGcTaHXfHtPJda9kDZ5uwqFUT6voH-8nRWG0/edit#heading=h.f2sa6ucbmwhr

File: web/src/components/HotTopic.js (hunk 1 above)
Comment (canyugs) [reply web/src/components/HotTopic.js:70]: It is useful for fake date XD

File: web/src/components/User.js (hunk 2 above)
Comment (canyugs) [reply web/src/components/User.js:245]: got It

**Pull request merged**